### PR TITLE
Security improvements

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-sdk-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [ ] **If this is a code change**: I have written unit tests.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Test Plan

Some tests have been updated to pass until near-workspaces's NAJ dependency is upgraded to v4.

### CODE REVIEW

File: packages/near-sdk-js/README.md
Comment (frol) [inline packages/near-sdk-js/README.md]: While I was reading this, I wondered if there is a good way to hint to users that certain host functions are deprecated. https://docs.rs/near-sdk/latest/near_sdk/env/fn.block_index.html

**Review by frol — APPROVED** [review]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@gtsonevv Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/gtsonevv?type=bot&pr=near/near-sdk-js/409&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/gtsonevv?type=bot&pr=near/near-sdk-js/409&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/gtsonevv)







<details><summary>Current status: <i>executed</i></summary>

| Reviewer | Score |
|--------|--------|
| @frol  | 8 |


Your contribution is much appreciated with a final score of 8!
You have received 100 (80 base + 10 weekly bonus + 10 monthly bonus) Sloth points for this contribution


Congratulations @gtsonevv! Your PR was highly scored and you completed another monthly streak! To keep your monthly streak make another pull request next month and get 8+ score for it
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/gtsonevv)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
### ❌ PR inactive
This PR has been inactive for two weeks. It's now marked as stale. 
>[!TIP]
If you think it's a mistake, you can restart the bot with `include` command to proceed 🔄

File: packages/near-sdk-js/README.md
Comment (gtsonevv) [reply packages/near-sdk-js/README.md]: Fixed it by adding a "Deprecated" label"

**Review by frol — APPROVED** [review]
@race-of-sloths score 8

**Pull request merged**
